xapian-core/weight/bm25weight.cc

```diff
@@ -35,11 +35,23 @@ using namespace std;
 
 namespace Xapian {
 
+BM25Weight::BM25Weight(bool& bm25+)
```

Comment:
I realise we're not keeping this constructor, but as a general point you don't pass parameters by non-`const` reference unless they are used to return a value - the parameter here should just be `bool`.

You wouldn't pass a primitive type like `bool` by `const` reference either - passing by `const` reference is a way to avoid copying the value, which is useful when copying is potentially expensive (e.g. for a `std::vector`, `std::map`, etc) but copying a `bool` (or `int`, `double`, etc) is cheap as it fits in a single CPU register, and dealing with the reference is likely to add some overhead so could well actually be slower).